compiler/daml-lf-ast/test/DA/Daml/LF/Ast/Tests.hs

```diff
@@ -59,3 +69,166 @@ substitutionTests = testGroup "substitution"
     beta11 = TypeVarName "beta11"
     vBeta1 = TVar beta1
     vBeta11 = TVar beta11
+
+
+typeSynTests :: TestTree
```

Comment:
It would be great if you could add one slightly bigger testcase that tests code a module that is more of the form that we will want to generate, e.g., a single module that defines type synonyms for `Functor` and `Applicative`, instances for one type and a definition that has an `Applicative` constraint. Don’t worry about exactly capturing how the generated code will look like.

Reply:
Not sure exactly what you want to see in this bigger example.
And what are we attempting to test, that isn't already tested?

Reply:
something like the following (in pseudocode)

```
type Functor f = [("fmap", forall a b. (a -> b) -> f a -> f b)]
type Applicative f = [("super", Functor f), ("pure", forall a. a -> f a), ("<*>", forall a b. f (a -> b) -> f b -> f a)]
optFunctor : Functor Optional
optFunctor = …
optApplicative : Applicative Optional
optApplicative = …
f : Applicative f -> a -> f a
f apDict = structProj "pure" apDict
```

We probably do test the individual pieces already but having a test that puts it altogether is helpful both for documentation purposes and to make sure that we didn’t forget anything.

Reply:
bigger example added

Reply:
it found two bugs, so thanks for asking for it

Reply:
Thanks a lot for adding it! It would be great if you could expand it to include the superclass case, i.e., `Applicative` (or `Pointed` or whatever) just to make sure there are no unforeseen issues there either.

Reply:
ok, done